debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/sink/NoStackTraceDlqException.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.sink;
+
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.sink.ErrantRecordReporter;
+
+import com.mongodb.lang.Nullable;
+
+/**
+ * An exception that is meant to be reported to the DLQ via {@link ErrantRecordReporter}. The
+ * {@linkplain Class#getName() class name} and {@linkplain #getMessage() message} must be treated as
+ * public API. This is necessary to allow the possibility of processing the DLQ in an automated way.
+ * The length of the {@linkplain #getStackTrace() stack trace} is always 0, which means that this
+ * exception is suitable for reporting only situations with the stack trace being inconsequential.
+ */
+abstract class NoStackTraceDlqException extends DataException {
```

Comment:
Why can't we keep the stack trace? Even if it seems to be inconsequential is most of the cases it can still be useful in others. What do we gain by dropping the stacktrace?

Reply:
This is meant for a custom error handling override to choose other options than the default behavior that comes for Kafka Connect sink connectors. I might remove this for now and file a Jira for adding it later. How is this going to be with Debezium Server for sink connectors? DLQs and sink error handling?

Reply:
That doesn't seem to answer the question though, does Kafka Connect mandate the use of `DataException` for these use cases or can it be any runtime exception ?

Reply:
@Naros Because you asked about that below. ;)

Reply:
I am going to remove this class for now. It is related to overriding the default Kafka Connect sink connector error handling with DLQ etc. I am going to file a Jira to add this later. It is not needed for the initial implementation. I hope to get it fully in but I think as my PTO comes up I can move this to a Jira and we can discuss and add it later. wdyt?

Reply:
I removed this class and related code for now (including `ErrorAnalyzer`). Those are meant to have a custom error handling overriding the default Kafka Connect DLQ/non-DLQ/error-tolerance error handling and giving more fine-constraint options to users. As those exception go as message into the DLQ the stack trace is about to be removed to not blow up message size. I will file a Jira for the overriding error handler behavior and keep the files locally until I re-add it again properly.